### PR TITLE
feat: first draft of copy dsl for pods.

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

**Comment by iocanel** [comment]
@rohanKanojia: I managed to solve the issue that we were having. It seems that the root cause was that the exec command was failing silently and thus nothing was written in the streams.

So, I removed the commit, with the experiments on the ExecWebSocketListener, timeouts etc, since it was not required for this to work.

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/PodOperationsImpl.java [hunk 1]
```diff
@@ -279,7 +289,242 @@ public ExecWatch exec(String... command) {
...
+        }
+      }.call();
+    } catch (NoClassDefFoundError e) {
+      throw new KubernetesClientException("Base64InputStream class is provided by commons-codec, an optional dependency. To use the read/copy functionality you must explicitly add this dependency to the classpath.");
+    }
+  }
+
+  //
+  //
+  // The copy and read utilities below have been inspired by Brendan Burns copy utilities on the offical kubernetes-client.
```
[the remark this answers is not in the document]
Comment (iocanel) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/PodOperationsImpl.java]: Yes

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/PodOperationsImpl.java [hunk 2]
```diff
@@ -279,7 +289,242 @@ public ExecWatch exec(String... command) {
...
+      }
+      throw new IllegalStateException("No file or dir has been specified");
+    } catch (Exception e) {
+      throw KubernetesClientException.launderThrowable(e);
+    }
+  }
+
+  private InputStream readFile(String source) throws Exception {
+    //Let's wrap the code to a callable inner class to avoid NoClassDef when loading this class.
+    try {
```
[the remark this answers is not in the document]
Comment (iocanel) [reply kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/PodOperationsImpl.java]: I think not!

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**